airbyte-integrations/bases/debezium/src/test/java/io/airbyte/integrations/debezium/internals/mongodb/MongoDbCdcTargetPositionTest.java

```diff
@@ -207,9 +211,12 @@ void testIsSameOffset() {
     when(changeStreamIterable.cursor()).thenReturn(mongoChangeStreamCursor);
     when(mongoClient.watch(BsonDocument.class)).thenReturn(changeStreamIterable);
 
-    final Map<String, String> offsetA = Map.of(MongoDbDebeziumConstants.OffsetState.VALUE_RESUME_TOKEN, RESUME_TOKEN);
-    final Map<String, String> offsetB = Map.of(MongoDbDebeziumConstants.OffsetState.VALUE_RESUME_TOKEN, RESUME_TOKEN);
-    final Map<String, String> offsetC = Map.of(MongoDbDebeziumConstants.OffsetState.VALUE_RESUME_TOKEN, "other resume token");
+    final Map<String, String> offsetA =
+        Jsons.object(MongoDbDebeziumStateUtil.formatState(null, null, RESUME_TOKEN), new TypeReference<>() {});
+    final Map<String, String> offsetB =
+        Jsons.object(MongoDbDebeziumStateUtil.formatState(null, null, RESUME_TOKEN), new TypeReference<>() {});
+    final Map<String, String> offsetC =
+        Jsons.object(MongoDbDebeziumStateUtil.formatState(null, null, OTHER_RESUME_TOKEN), new TypeReference<>() {});
```

Comment:
@jpefaur Not sure, but maybe that amount of documents causes the resume token stored in the state to become invalid, forcing an initial snapshot.  We do have a test for the target position, so I can add a test there that handles this case instead of trying to guess if adding a bunch of documents covers it.